packages/workflow/src/records/handler/download.ts

```diff
@@ -54,6 +56,15 @@ export async function downloadRecordHandler(
   const token = getToken(request)
   // Task history is fetched rather than the task only
   const record = await getValidRecordById(payload.id, token, true)
+  const assignment = findAssignment(record)
+
+  if (assignment) {
+    const userOrSystem = await getUserOrSystem(token)
+    const practitionerId = userOrSystem.practitionerId
+
+    if (assignment.practitioner.id !== practitionerId)
+      throw new Error('Record is assigned to a different user')
+  }
```

Comment:
We could consider having `practitionerId` in the JWT as all systems and users have it 🤔

Reply:
Nice!